Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -68,25 +67,26 @@ extension MLDSA {
         static let bytesCount = Backing.bytesCount
 
         fileprivate final class Backing {
-            private let pointer: UnsafeMutablePointer<MLDSA65_private_key>
-            private let seedPointer: UnsafeMutablePointer<UInt8>
-
-            func withUnsafePointer<T>(_ body: (UnsafePointer<MLDSA65_private_key>) throws -> T) rethrows -> T {
-                try body(self.pointer)
-            }
+            var key: MLDSA65_private_key
+            var seed: Data
 
             /// Initialize a ML-DSA-65 private key from a random seed.
             init() throws {
-                self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
-                self.seedPointer = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.seedSizeInBytes)
-
-                try withUnsafeTemporaryAllocation(of: UInt8.self, capacity: MLDSA.PublicKey.Backing.bytesCount) { publicKeyPtr in
-                    guard CCryptoBoringSSL_MLDSA65_generate_key(
-                        publicKeyPtr.baseAddress,
-                        self.seedPointer,
-                        self.pointer
-                    ) == 1 else {
-                        throw CryptoKitError.internalBoringSSLError()
+                (self.key, self.seed) = try withUnsafeTemporaryAllocation(of: MLDSA65_private_key.self, capacity: 1) { privateKeyPtr in
```

Comment:
I don't think we need an `unsafeTemporaryAllocation`. We can just zero-initialize the value `self.key = .init()` and then take a pointer to it.

Reply:
Got it. I also tried to remove the seed's `unsafeTemporaryAllocation` by passing to the BoringSSL function an empty `[UInt8]` var and then initializing the `Data` object from it, but I encountered some seg faults in testing